File: _code_examples/kamon-1.0.x/kamon-core-examples/src/main/java/kamon/examples/java/CustomConfiguration.java
package kamon.examples.java;

import com.typesafe.config.Config;
import com.typesafe.config.ConfigFactory;
import kamon.Kamon;

// tag:custom-configuration:start
public class CustomConfiguration {
  public static void main(String[] args) {
    final Config customConfig = ConfigFactory.load("custom-config");
    final Config codeConfig = ConfigFactory.parseString(
      "kamon.metric.track-unmatched-entities = no");

    Kamon.start(codeConfig.withFallback(customConfig));
    // Kamon gets initialized with the provided configuration rather
    // than calling ConfigFactory.load() itself.


    // This application wont terminate unless you shutdown Kamon.
    Kamon.shutdown();
  }
}
// tag:custom-configuration:end

File: examples/kamon-core-examples/src/main/scala/kamon/examples/scala/CustomConfiguration.scala
package kamon.examples.scala

import com.typesafe.config.ConfigFactory
import kamon.Kamon

// tag:custom-configuration:start
object CustomConfiguration {
  val customConfig = ConfigFactory.load("custom-config")
  val codeConfig = ConfigFactory.parseString(
    "kamon.metric.track-unmatched-entities = no")

  Kamon.start(codeConfig.withFallback(customConfig))
  // Kamon gets initialized with the provided configuration rather
  // than calling ConfigFactory.load() itself.


  // This application wont terminate unless you shutdown Kamon.
  Kamon.shutdown()
}
// tag:custom-configuration:end
